Integrations/integration-Panorama.yml

```diff
@@ -0,0 +1,148 @@
+detaileddescription: |-
+  The integration uses the Panorama XML API.
+  To obtain an API Key, use the following REST command:
+  https://[PanoramaIP]/api/?type=keygen&user=[user]&password=[password]
+  and copy the resulting key.
+  For more information, visit: https://www.paloaltonetworks.com/documentation
+configuration:
+- display: Server URL (e.g. https://192.168.0.1)
+  name: server
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: API Key
+  name: key
+  defaultvalue: ""
+  type: 4
+  required: true
+- display: Port
+  name: port
+  defaultvalue: ""
+  type: 0
+  required: false
+- display: Do not validate server certificate (insecure)
+  name: insecure
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: Use system proxy settings
+  name: proxy
+  defaultvalue: ""
+  type: 8
+  required: false
+script:
+  script: |-
+    var sendRequest = function(url) {
+        url.replace(/[\/]+$/, '');
+        var res = http(
+            url,
+            {
+                Method: 'POST',
+                Headers: {'Content-Type': ['application/x-www-form-urlencoded']},
+            },
+            params.insecure,
+            params.proxy
+        );
+
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            throw 'Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+        }
+
+        return x2j(res.Body);
+    };
+
+    var sendMultipartRequest = function(url, file, apikey){
+            var res = httpMultipart(
+              url,
+              file,
+              {
+                Method: 'POST',
+              },
+              {
+                'key': apikey,
+                'file' : file
+              },
+              params.insecure,
+              params.proxy
+            );
+
+            if (res.StatusCode !==200) {
+              throw 'Request Failed.\nStatus code: ' + res.StatusCode +' '+errorDict[res.StatusCode]+ '.\nBody: ' + JSON.stringify(res) + '.';
+            }
+            return JSON.parse(x2j(res.Body));
```

Comment:
we also had a case where the response was 200, but the xml itself has an error. So we need to check this scenario as well and throw